docs/advanced/misc.rst

```diff
@@ -153,6 +153,72 @@ following checklist.
   within pybind11 that will throw exceptions on certain GIL handling errors
   (reference counting operations).
 
+Free-threading support
+==================================================================
+
+pybind11 supports the experimental free-threaded Python build.  pybind11's internal data
+structures are thread safe. To enable your modules to be used with free-threading, pass the
+:class:`mod_gil_not_used` tag as the third argument to ``PYBIND11_MODULE``.
+
+For example:
+
+.. code-block:: cpp
+    :emphasize-lines: 1
+    PYBIND11_MODULE(example, m, py::mod_gil_not_used()) {
+        py::class_<Animal> animal(m, "Animal");
+        // etc
+    }
+
+Note, of course, enabling your module to be used in free threading is also your promise that
+your code is thread safe. Modules must still be built against the Python free-threading branch to enable
+free-threading, even if they specify this tag.  Adding this tag does not break compatibility with non-free-threaded
+Python.
+
+Sub-interpreter support
+==================================================================
+
+pybind11 supports isolated sub-interpreters.  Pybind11's internal data structures are
+sub-interpreter safe. To enable your modules to be imported in isolated sub-interpreters, pass the
+:func:`multiple_interpreters::per_interpreter_gil()` tag as the third or later argument to
+``PYBIND11_MODULE``.
+
+For example:
+
+.. code-block:: cpp
+    :emphasize-lines: 1
+    PYBIND11_MODULE(example, m, py::mod_gil_not_used(), py::multiple_interpreters_per_interpreter_gil()) {
```

Comment:
```
py::multiple_interpreters::per_interpreter_gil()
```

Suggestion to not list `py::mod_gil_not_used()` here, since people will wonder "why"?

Instead, add to the last paragraph ("Note: Sub-interpreter support does not imply free-threading support or vice-versa."), to explain that `py::multiple_interpreters::...` can be combined with `py::mod_gil_not_used()`, and maybe clarify what the various combinations mean (it's still not entirely clear to me, TBH).

Reply:
Would be nice to have an example somewhere, though, I think most extensions really want to have both enabled unless they don’t support it.

Reply:
I added what turned out to be a rather long section that hopefully explains the differences well (with an example).  Let me know what you think...